### PR TITLE
add `NatSpec`

### LINKED ISSUE #10: add `NatSpec` comments

### PR DESCRIPTION
**Pull request by YouStillAlive**
closes #10

### CODE REVIEW

File: contracts/DispenserInternal.sol [hunk 1]
```diff
@@ -61,6 +94,11 @@ abstract contract DispenserInternal is IDispenserProvider, DealProvider, Dispens
         lockDealNFT.cloneVaultId(poolId, tokenPoolId);
     }
 
+    /// @notice Withdraws tokens from the provider if the withdrawable amount is greater than zero.
+    /// @dev Transfers the tokens from the owner to the `lockDealNFT` contract if available.
+    /// @param provider The simple provider from which tokens will be withdrawn.
+    /// @param poolId The unique identifier for the pool to withdraw from.
+    /// @param owner The address of the owner of the tokens being withdrawn.
     function _withdrawIfAvailable(
         ISimpleProvider provider,
```
Comment (Lomet) [inline contracts/DispenserInternal.sol:103]: why do we need the provider here?

**Review by Lomet — APPROVED** [review]
Looks good ?

File: contracts/DispenserInternal.sol (hunk 1 above)
Comment (YouStillAlive) [reply contracts/DispenserInternal.sol:107]: yes, we can use `lockDealNFT.getWithdrawableAmount` instead of `provider.getWithdrawableAmount`

Comment (YouStillAlive) [reply contracts/DispenserInternal.sol:107]: let's do this in another PR

**Pull request merged**